python-mini-projects/projects/Terminal_progress_bar_with_images_resizing/progress_bar_ with_images_resizing.py

```diff
@@ -0,0 +1,30 @@
+from tqdm import tqdm
+from PIL import Image
+import os
+from time import sleep
+
+
+def Resize_image(size, image):
+    if os.path.isfile(image):
+        try:
+            im = Image.open(image)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *im*. The resource is allocated by call *Image.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *im* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).